Sources/CoreLocationMiddleware/CoreLocationMiddleware.swift

```diff
@@ -1,40 +1,59 @@
 import CoreLocation
 import SwiftRex
 
-public struct LocationState: Equatable {
-    var authzType: AuthzType
-    var authzStatus: CLAuthorizationStatus
-    var location: CLLocation
-    
-    public init(authzType: AuthzType = .whenInUse,
-                authzStatus: CLAuthorizationStatus,
-                location: CLLocation) {
-        self.authzType = authzType
-        self.authzStatus = authzStatus
-        self.location = location
-    }
-}
 
+// MARK: - ACTION
 public enum LocationAction {
     // Input
+    case request(RequestAction)
+    // Output
+    case status(StatusAction)
+}
+
+public enum RequestAction {
     case startMonitoring
     case stopMonitoring
     case requestAuthorizationStatus
     case requestAuthorizationType
     case requestPosition
     case requestDeviceCapabilities
-    // Output
+}
+
+public enum StatusAction {
     case gotPosition(CLLocation)
-    case gotAuthzStatus(CLAuthorizationStatus)
+    case gotAuthzStatus(AuthzStatus)
     case gotDeviceCapabilities(DeviceCapabilities)
     case receiveError(Error)
 }
 
+// MARK: - STATE
+public struct LocationState: Equatable {
+    var authzType: AuthzType
+    var authzStatus: CLAuthorizationStatus
+    var authzAccuracy: CLAccuracyAuthorization?
+    var location: CLLocation
+    
+    public init(authzType: AuthzType = .whenInUse,
+                authzStatus: CLAuthorizationStatus,
+                authzAccuracy: CLAccuracyAuthorization? = .none,
+                location: CLLocation) {
+        self.authzType = authzType
+        self.authzStatus = authzStatus
+        self.authzAccuracy = authzAccuracy
+        self.location = location
+    }
+}
+
 public enum AuthzType: Equatable {
     case whenInUse
     case always
 }
 
+public struct AuthzStatus: Equatable {
+    public let status : CLAuthorizationStatus
+    public let accuracy: CLAccuracyAuthorization?
+}
+
```

Comment:
New struct to compile both CLAuthorizationStatus and CLAccuracyAuthorization. Note that I used an Optional here because this info isn't available prior to iOS 14